### PR TITLE
Add way to get dynamic counters directly from the stats object

### PR DESCRIPTION
**Pull request by mdordal**
It's sometimes useful to be able to get dynamic counters without
constructing the entire stats map. Add a way to ask the Stats
object for the counter directly.

Test Plan: Add a unit test

### CODE REVIEW

File: stats/src/main/java/com/facebook/stats/mx/StatsReader.java [hunk 1]
```diff
@@ -36,6 +36,8 @@
   public MultiWindowDistribution getDistribution(String key);
   public String getAttribute(StatType key);
   public String getAttribute(String key);
+  public Callable<Long> getDynamicCounter(StatType key);
+  public Callable<Long> getDynamicCounter(String key);
```
Comment (srash) [inline stats/src/main/java/com/facebook/stats/mx/StatsReader.java]: what's the use case for wanting the callable versus just the value?  what's the difference between

  stats.getAttribute(key)
vs
  stats.getDynamicCounter(key).call()

when would you want to be responsible for call() directly versus just asking the stats object for the current value?

further, as the user of the stats object, you will be setting the callable...if you need it outside the context of the stats object, why not save it?

i guess i don't know the use case

Comment (srash) [inline stats/src/main/java/com/facebook/stats/mx/StatsReader.java]: and i don't think the stats object should be used as some sort of global map to share callables across

Comment (mdordal) [inline stats/src/main/java/com/facebook/stats/mx/StatsReader.java]: I don't have any need for the Callable<> directly, but the other getFooCounter things returned the underlying object. Would you prefer it just return the Long?

Comment (srash) [inline stats/src/main/java/com/facebook/stats/mx/StatsReader.java]: i didn’t realize there wasn’t any way at all to get a dynamic coutner directly (diff name makes sense now)

i think returning the counter value is more consistent with the attribute API

also, i sort of think that making get/add counter work like they are in the same hash (ie same namespace)

not sure how hard this is offhand: we’d need to unify the two counter hashes or do a check if they exist in both first (requires locking, bleah)

probably this way (either with the Callable or just the value) is best

**Pull request merged**